controllers/cruisecontroltask_controller.go

```diff
@@ -195,55 +309,71 @@ func (r *CruiseControlTaskReconciler) UpdateStatus(ctx context.Context, instance
 			if err != nil {
 				return errors.WithMessage(err, "failed to get updated Kafka Cluster CR before updating its status")
 			}
 			taskAndStates.SyncState(instance)
-			err = r.Client.Status().Update(ctx, instance)
-			if err != nil {
-				return err
-			}
-		} else {
-			return err
 		}
+		return err
+	}); err != nil {
+		return err
 	}
+
 	return nil
 }
 
 // SetupCruiseControlWithManager registers cruise control controller to the manager
 func SetupCruiseControlWithManager(mgr ctrl.Manager) *ctrl.Builder {
-	builder := ctrl.NewControllerManagedBy(mgr).
-		For(&kafkav1beta1.KafkaCluster{}).
-		WithEventFilter(SkipClusterRegistryOwnedResourcePredicate{}).
-		Named("CruiseControl")
+	cruiseControlOperationPredicate := predicate.Funcs{
+		// We dont reconcile when there is no operation state
+		CreateFunc: func(e event.CreateEvent) bool {
+			obj := e.Object.(*banzaiv1alpha1.CruiseControlOperation)
+			return obj.GetCurrentTaskState() != ""
+		},
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			oldObj := e.ObjectOld.(*banzaiv1alpha1.CruiseControlOperation)
+			newObj := e.ObjectNew.(*banzaiv1alpha1.CruiseControlOperation)
+			if !reflect.DeepEqual(oldObj.GetCurrentTask(), newObj.GetCurrentTask()) || oldObj.IsPaused() != newObj.IsPaused() ||
+				oldObj.GetDeletionTimestamp() != newObj.GetDeletionTimestamp() ||
+				oldObj.GetGeneration() != newObj.GetGeneration() {
+				return true
+			}
+			return false
+		},
+	}
 
-	builder.WithEventFilter(
-		predicate.Funcs{
-			UpdateFunc: func(e event.UpdateEvent) bool {
-				if _, ok := e.ObjectNew.(*kafkav1beta1.KafkaCluster); ok {
-					oldObj := e.ObjectOld.(*kafkav1beta1.KafkaCluster)
-					newObj := e.ObjectNew.(*kafkav1beta1.KafkaCluster)
-					if !reflect.DeepEqual(oldObj.Status.BrokersState, newObj.Status.BrokersState) ||
-						oldObj.GetDeletionTimestamp() != newObj.GetDeletionTimestamp() ||
-						oldObj.GetGeneration() != newObj.GetGeneration() {
-						return true
-					}
-					return false
+	kafkaClusterPredicate := predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			if _, ok := e.ObjectNew.(*banzaiv1beta1.KafkaCluster); ok {
+				oldObj := e.ObjectOld.(*banzaiv1beta1.KafkaCluster)
+				newObj := e.ObjectNew.(*banzaiv1beta1.KafkaCluster)
+				if !reflect.DeepEqual(oldObj.Status.BrokersState, newObj.Status.BrokersState) ||
+					oldObj.GetDeletionTimestamp() != newObj.GetDeletionTimestamp() ||
+					oldObj.GetGeneration() != newObj.GetGeneration() {
+					return true
 				}
-				return true
-			},
-		})
+				return false
+			}
+			return true
+		},
+	}
 
-	return builder
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&banzaiv1beta1.KafkaCluster{}).
+		WithEventFilter(SkipClusterRegistryOwnedResourcePredicate{}).
+		WithEventFilter(kafkaClusterPredicate).
+		Owns(&banzaiv1alpha1.CruiseControlOperation{}, builder.WithPredicates(cruiseControlOperationPredicate)).
+		Named("CruiseControl")
```

Comment:
What about having this to match the controller name?
```suggestion
		Named("CruiseControlTask")
```